ui/app/components/app/transaction-list/transaction-list-hooks.component.js

```diff
@@ -0,0 +1,161 @@
+import React, { useCallback, useMemo, useEffect, useRef } from 'react'
+import PropTypes from 'prop-types'
+import { useSelector, useDispatch } from 'react-redux'
+import {
+  nonceSortedCompletedTransactionsSelector,
+  nonceSortedPendingTransactionsSelector,
+} from '../../../selectors/transactions'
+import {
+  getSelectedAddress,
+  getAssetImages,
+  getFeatureFlags,
+} from '../../../selectors/selectors'
+import { selectedTokenSelector } from '../../../selectors/tokens'
+import * as actions from '../../../ducks/gas/gas.duck'
+import { useI18nContext } from '../../../hooks/useI18nContext'
+import TransactionListItem from '../transaction-list-item'
+
+
+const mapStateToProps = (state) => {
+  const pendingTransactions = nonceSortedPendingTransactionsSelector(state)
+  const firstPendingTransactionId =
+    pendingTransactions[0] && pendingTransactions[0].primaryTransaction.id
+  return {
+    completedTransactions: nonceSortedCompletedTransactionsSelector(state),
+    pendingTransactions,
+    firstPendingTransactionId,
+    selectedToken: selectedTokenSelector(state),
+    selectedAddress: getSelectedAddress(state),
+    assetImages: getAssetImages(state),
+    transactionTimeFeatureActive: getFeatureFlags(state).transactionTime,
+  }
+}
+
+function useTransactionListState () {
```

Comment:
This hook encapsulates all of the redux state selection as well as the mount/update effects from .component